# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBObjectStore.cpp
# Commit: d6aa1c5bc7d9
# Full Hash: d6aa1c5bc7d96f6a9bc0a76508c50cc852cf3f2e
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-10 15:51:07
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Change definition of OpenCursorParams subtypes to allow reducing duplicated code r=ttung
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40955
# ==============================================================================

diff -r 6e46dad972c8 -r d6aa1c5bc7d9 dom/indexedDB/IDBObjectStore.cpp
--- a/dom/indexedDB/IDBObjectStore.cpp	Tue Sep 10 16:47:08 2019 +0300
+++ b/dom/indexedDB/IDBObjectStore.cpp	Tue Sep 10 16:47:26 2019 +0300
@@ -2273,16 +2273,16 @@
   OpenCursorParams params;
   if (aKeysOnly) {
     ObjectStoreOpenKeyCursorParams openParams;
-    openParams.objectStoreId() = objectStoreId;
-    openParams.optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.direction() = direction;
+    openParams.commonParams().objectStoreId() = objectStoreId;
+    openParams.commonParams().optionalKeyRange() = std::move(optionalKeyRange);
+    openParams.commonParams().direction() = direction;
 
     params = std::move(openParams);
   } else {
     ObjectStoreOpenCursorParams openParams;
-    openParams.objectStoreId() = objectStoreId;
-    openParams.optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.direction() = direction;
+    openParams.commonParams().objectStoreId() = objectStoreId;
+    openParams.commonParams().optionalKeyRange() = std::move(optionalKeyRange);
+    openParams.commonParams().direction() = direction;
 
     params = std::move(openParams);
   }